src/main/java/org/quantumbadger/redreader/http/okhttp/OKHTTPBackend.java

```diff
@@ -31,11 +35,26 @@
 public class OKHTTPBackend implements HTTPBackend {
 
 	private final OkHttpClient mClient;
+	private static HTTPBackend httpBackend;
+	public static boolean reset = false;
+
+	public static HTTPBackend getHttpBackend(boolean force) {
+		if(force || httpBackend == null) {
+			httpBackend = new OKHTTPBackend();
+			reset = false;
+		}
+		return httpBackend;
+	}
 
 	public OKHTTPBackend() {
 
 		mClient = new OkHttpClient();
 
+		if(BaseActivity.getTorStatus()) {
+			Proxy tor = new Proxy(Proxy.Type.HTTP, new InetSocketAddress("127.0.0.1", 8118));
```

Comment:
Maybe a silly question, but do HTTP proxies handle HTTPS requests? Either way it might be better to use SOCKS here (I think Orbot and OkHTTP both support it).